student-os/scripts/group_git_changes.py

```
@@ -184,6 +184,34 @@ def staged_blob_size(repo: Path, path: str) -> int | None:
         return None
 
 
+def git_diff_quiet(repo: Path, path: str, *, cached: bool = False, ignore_cr_at_eol: bool = False) -> bool:
+    command = ["git", "-C", str(repo), "diff", "--quiet"]
+    if cached:
+        command.append("--cached")
+    if ignore_cr_at_eol:
+        command.append("--ignore-cr-at-eol")
+    command.extend(["--", path])
+    result = subprocess.run(command, check=False, capture_output=True)
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)